omnisend/includes/SDK/V1/class-event.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Omnisend Client
+ *
+ * @package OmnisendClient
+ */
+
+namespace Omnisend\SDK\V1;
+
+use Omnisend\SDK\V1\Contact;
+use WP_Error;
+
+defined( 'ABSPATH' ) || die( 'no direct access' );
+
+/**
+ * Omnisend Event class. It's should be used with Omnisend Client.
+ *
+ */
+class Event {
+
+
+	private $contact          = null;
+	private $event_name       = null;
+	private $event_time       = null;
+	private $origin           = null;
+	private array $properties = array();
+
+	/**
+	 * Validate event properties.
+	 *
+	 * TODO kas yra reuquired? ka reikia vailiduoti?
+	 * It ensures that all required properties are set
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+
+		$error = new WP_Error();
+		if ( $contact instanceof Contact ) {
+			$error->merge_from( $contact->validate() );
+		}
```

Comment:
I guess else part is missing?

Reply:
Also more validation is needed. Event name, origin (if is it mandatory).

Reply:
added more validation, please check

Reply:
Accessing class variable should be without $
$this->$event_name -> $this->event_name
Please check all places.